collect_app/src/main/java/org/odk/collect/android/activities/SplashScreenActivity.java

```diff
@@ -214,7 +213,7 @@ public void run() {
 
     private void createErrorDialog(String errorMsg, final boolean shouldExit) {
         Collect.getInstance().getActivityLogger().logAction(this, "createErrorDialog", "show");
-        alertDialog = new AlertDialog.Builder(this).create();
+        AlertDialog alertDialog = new AlertDialog.Builder(this).create();
         alertDialog.setIcon(android.R.drawable.ic_dialog_info);
         alertDialog.setMessage(errorMsg);
         DialogInterface.OnClickListener errorListener = new DialogInterface.OnClickListener() {
```

Comment:
Not anything major but always good to simplify most of these listeners when you stumble upon them as we try to simplify the code with lambda.

Reply:
I think it would be better to introduce lambdas wherever it's possible in a separate pr probably there is a separate check for it, this one is for something else so I don't like mixing different improvements.

Reply:
This is largely a matter of taste but I would prefer that the changes be made in a PR like this one that touches the code for other reasons but as a separate commit. That way we can more easily visually inspect the change and make sure it goes through regression testing. Big PRs that make structural changes like this in a lot of places are really hard to review.

Reply:
That said, this PR is already a refactor PR so yes, probably better to keep it cleanly focused on one thing. When there are deeper code modifications then I think it's nice to have some cleanup commits.